dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -1287,13 +1288,20 @@ protected List<Node> convertCallArguments(
             } else {
                 assert lastParamType instanceof ArrayType
                         : "variable argument formal must be an array";
-                // Apply method invocation conversion to lastArgIndex arguments and use the
-                // remaining ones to initialize an array.
-                for (int i = 0; i < lastArgIndex; i++) {
-                    Node actualVal = scan(actualExprs.get(i), null);
-                    convertedNodes.add(methodInvocationConvert(actualVal, formals.get(i)));
+                // For an inner class constructor, add the receiver as the first arguments
```

Comment:
This seems like an odd place for this change... it's an else branch where some sizes mismatched. Shouldn't this adaptation happen before we check the sizes?

Reply:
Thanks! Yes, moving it even before if (method.isVarArgs()) also makes sense.